lib/echo_server.rb

```diff
@@ -1,2 +1,63 @@
+require 'socket'
+require_relative 'server_socket'
+require_relative 'echo_response'
+
+
 class EchoServer
+    attr_reader :server, :is_server_open, :disconnect
+
+    def initialize(server_socket)
+        @server = nil
+        @server_socket = server_socket
+        @is_server_open = false
+        @is_connected = false
+    end
+
+    def start
+        open_server_connection
+        accept_incoming_connections
+    end
+
+    def open_server_connection
+        @server = @server_socket.open
+        @is_server_open = true
+        puts "Server is listening on port #{@server_socket.port}"
+    end
+
+    def client_handler(incoming_connection) 
+        Thread.new(incoming_connection) do |connected|
+            @is_connected = true
```

Comment:
Many clients can connect to the server, but this variable doesn't take that into account necessarily. 

What happens if 2 clients connect to the server, then 1 disconnects. What's the value of `is_connected` ? 

It seems like it would be `false`. Also, it seems like is_connected might be poorly named, and/or points to a smell that this may be the wrong data structure